File: web/src/app/models/permission.ts
import { Season } from "./season";
import { User } from "./user";
import { PermissionType } from "./permission-type";

export class Permission {
  constructor(
    public id: string,
    public user: User,
    public season: Season,
    public seasonId: string,
    public type: PermissionType,
    public userId: string,
    public username?: string,
  ) {}
}

File: api/Models/Permission.cs
using System.Text.Json.Serialization;

namespace api.Models
{
    public class Permission
    {
        public Guid Id { get; set; }
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual User User { get; set; } = default!;
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public virtual Season Season { get; set; } = default!;
        public Guid UserId { get; set; }
        public Guid SeasonId { get; set; }
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public PermissionType Type { get; set; }
    }

    public enum PermissionType
    {
        Moderator,
        Admin
    }
}
